Fix remaining broken tests in manage_treemap Django app

fungjj92 opened the pull request:
## Overview

After #3288 , we decided to re-strategize test-fixing work by targeting a django-app at a time. This PR targets fixing the failing tests in `manage_treemap` but funny enough just touches files in the `treemap` folder. No matter.

The main takeaways are:
- Images, like csv's, want to be byte streams in py3
- Modgrammar had breaking changes that were unimplemented in OTM in the recent version bump

I am not certain but the modgrammar issues may affect other apps. If so, be aware as to not duplicate effort.

### Notes
This branch works off #3288.

Huge drop off in test errors in the project after these fixes 🎉 
```
Ran 1115 tests in 339.283s

FAILED (failures=25, errors=48, skipped=23)
```

### Testing

Test that this app's tests are fixed
`./scripts/manage.sh test manage_treemap`

Pleeeaseee ensure that the modgrammar parsing changes make sense. I don't have enough context to fully understand if I fixed the code to fix the tests or I fixed the code to reinstate the desired functionality. Theoretically those two things should be the same, if test coverage is good and robust, but double check me please.

fungjj92 committed:
Encode image data in bytes not string

Refs https://github.com/OpenTreeMap/otm-cloud/issues/505

fungjj92 committed:
Upgrade modgrammar behavior for v.10

Some breaking changes in usage in modgrammar v.9 => v.10. Whitespace cleaning is invoked through a different class property (bool: grammar_whitespace => str: grammar_whitespace_mode) and  stripping is no longer done by default. The parse_string method has strictly defined behavior, prefer to use parse_text with args set.

Modgrammar is no longer maintained (or so it seems) and the latest / most helpful documentation is this google group post: https://groups.google.com/forum/#!msg/modgrammar/3CsPBb_UCck/kKr-0YmwtJ8J

jwalgran commented:
Looking at this now

jwalgran commented:
Actually looking at #3288 first, then this right after.

fungjj92 committed:
Handle image data as bytes not unicode

fungjj92 committed:
Encode all strings to bytes before hashing

jwalgran commented:
Looking at this now.

jwalgran commented:
The link to the modgrammer Google Group included in the commit message has an extra slash. This link worked for me

https://groups.google.com/forum/#!msg/modgrammar/3CsPBb_UCck/kKr-0YmwtJ8J

jwalgran reviewed (commented):
I looked at the last 2 commits, which appear to be the new ones added on top of `jf/fix-tests-2to3`. Nice work.

I'm thankful that we still have that well written 0.10 release post from 2013. I read through it closely and the changes you made to our `Grammar` subclasses look like the correct ones to maintain the previous behavior.

Waiting to give a final approval until the conflict is resolved.

fungjj92 commented:
Thanks for the attention to detail in your review! I've rebased successfully and updated the google groups post link.

jwalgran commented on opentreemap/api/tests.py:
```diff
@@ -373,7 +373,8 @@ def test_locations_plots_endpoint_max_plots_param_must_be_a_number(self):
             self.client,
             "%s/instance/%s/locations/0,0/plots?max_plots=foo" % (
-                API_PFX, self.instance.url_name))
+                API_PFX, self.instance.url_name)
```
It looks like this change has been reduced to moving a parenthesis. Consider removing it.

fungjj92 replied:
Good catch. I've dropped the associated commit where this change stood alone.

jwalgran reviewed (approved):
Confirmed. Nice work.

fungjj92 committed:
Format request params as unicode strings

fungjj92 committed:
Repackage bytestream into string stream for csv

A CSV by python definition is an iterator object that returns a string. A python3 string is defined as unicode and not binary, explicitly, so we have to be explicit about passing string data to csv generators.
A downside to this solution is we may now have 2 copies of the entire file in memory while it is being sliced into individual database rows but this should probably be ok given the expected use cases and available computing resources.

fungjj92 committed:
Remove obsolete unicode literal

Strings are implicitly unicode in py3

fungjj92 commented:
Thank you for this cascade of reviews!!

Ran tests to verify the cascade of PR's is working as planned. Verified only `treemap` tests are failing. Merging.
```
Ran 1121 tests in 318.913s

FAILED (failures=18, errors=11, skipped=23)
```
